### PR TITLE
feat(abstract-eth): add support to build tx for CCR of unsupported token

### PR DESCRIPTION
**Pull request by kamleshmugdiya**
Ticket: COIN-742

### CODE REVIEW

File: modules/abstract-eth/src/lib/transferBuilder.ts [hunk 1]
```diff
@@ -101,6 +101,11 @@ export class TransferBuilder {
     throw new InvalidParameterValueError('Invalid expiration time');
   }
 
+  tokenContractAddress(tokenContractAddress?: string): TransferBuilder {
```
Comment (margueriteblair) [inline modules/abstract-eth/src/lib/transferBuilder.ts:104]: Why have `tokenContractAddress` be optional?

File: modules/sdk-coin-opeth/test/unit/opeth.ts [hunk 2]
```diff
@@ -778,5 +781,68 @@ describe('Optimism', function () {
...
+        bitgoFeeAddress: bitgoFeeAddress,
+        recoveryDestination: destinationAddress,
+        eip1559: { maxFeePerGas: 20000000000, maxPriorityFeePerGas: 10000000000 },
+        gasLimit: 500000,
+        bitgoDestinationAddress: bitgoDestinationAddress,
+        tokenContractAddress: tokenContractAddress,
+      })) as OfflineVaultTxInfo;
+
+      should.exist(transaction);
+      transaction.should.have.property('txHex');
```
Comment (mullapudipruthvik) [inline modules/sdk-coin-opeth/test/unit/opeth.ts:824]: build the transaction from txHex and see whether tokenContractAddress is defined

File: modules/abstract-eth/src/abstractEthLikeNewCoins.ts [hunk 3]
```diff
@@ -1777,7 +1783,7 @@ export abstract class AbstractEthLikeNewCoins extends AbstractEthLikeCoin {
     const response: OfflineVaultTxInfo = {
       txHex: tx.toBroadcastFormat(),
       userKey,
-      coin: token,
+      coin: token ? token : this.getChain(),
```
Comment (mullapudipruthvik) [inline modules/abstract-eth/src/abstractEthLikeNewCoins.ts:1786]: please test this change for hot and cold wallets before merging

Comment (kamleshmugdiya) [reply modules/abstract-eth/src/abstractEthLikeNewCoins.ts:1786]: tested for hot wallet (supported and unsupported token).
will test for cold as well.

Comment (kamleshmugdiya) [reply modules/abstract-eth/src/abstractEthLikeNewCoins.ts:1786]: tested for cold wallet unsupported token. it's working as expected.

File: modules/sdk-coin-opeth/test/unit/opeth.ts (hunk 2 above)
Comment (kamleshmugdiya) [reply modules/sdk-coin-opeth/test/unit/opeth.ts:826]: built tx from txHex and added check that data should start with `0x0dcd7a6c`

**Review by mullapudipruthvik — APPROVED** [review]

**Pull request merged**